src/coreclr/vm/appdomain.cpp

```diff
@@ -3337,7 +3337,9 @@ void AppDomain::AddUnmanagedImageToCache(LPCWSTR libraryName, NATIVE_LIBRARY_HAN
 
     size_t len = (wcslen(libraryName) + 1) * sizeof(WCHAR);
     AllocMemHolder<WCHAR> copiedName(GetLowFrequencyHeap()->AllocMem(S_SIZE_T(len)));
-    memcpy(copiedName, libraryName, len);
+    if (len && libraryName) {
```

Comment:
This should not be needed. We are always copying non-empty block here.

Reply:
We also don't need to check for `libraryName` as it was already validated with the call to `wcslen`.